src/d/actor/d_a_majuu_flag.cpp

```diff
@@ -6,55 +6,794 @@
+            13,           8,           7,
+        },
+        {
+            8,           12,          18,          19,
+            14,           9,           8,
+        },
+        {
+            9,           13,          19,          20,
+        -1,    -1,    -1,
+        },
+        {
+            16,           10,    -1,    -1,
+        -1,    -1,    -1,
+        },
+        {
+            17,           11,           10,           15,
+        -1,    -1,    -1,
+        },
+        {
+            18,           12,           11,          16,
+        -1,    -1,    -1,
+        },
+        {
+            19,           13,           12,          17,
+        -1,    -1,    -1,
+        },
+        {
+            20,           14,           13,          18,
+        -1,    -1,    -1,
+        },
+        {
+            14,          19,    -1,    -1,
+        -1,    -1,    -1,
+        },
+    };
+    cXyz sp80;
+    cXyz sp74;
+    cXyz sp68;
+    cXyz sp5C;
+    cXyz sp50;
+    cXyz* pos = getPos();
+
+    sp50 = pos[index];
+
+    sp5C.setall(0.0f);
+    s32* relPos = rel_pos_idx_tbl[index];
+
+    for (s32 i = 0; i < 6; i++, relPos++) {
+        if (relPos[1] == -1) {
+            break;
+        }
+
+        sp80 = pos[relPos[0]] - sp50;
+        sp74 = pos[relPos[1]] - sp50;
+        sp68 = sp74.outprod(sp80);
+        sp68 = sp68.normZC();
+
+        sp5C += sp68;
+    }
+
+    sp5C = sp5C.normZC();
+
+    s32 iVar2;
+    s32 iVar4;
+    if (index == 0) {
+        iVar4 = 0;
+        iVar2 = 0;
+    } else if (index < 3) {
+        iVar4 = 1;
+        iVar2 = index + -1;
+    } else if (index < 6) {
+        iVar4 = 2;
+        iVar2 = index + -3;
+    } else if (index < 10) {
+        iVar4 = 3;
+        iVar2 = index + -6;
+    } else if (index < 0xf) {
+        iVar4 = 4;
+        iVar2 = index + -10;
+    } else {
+        iVar4 = 5;
+        iVar2 = index + -0xf;
+    }
+
+    MtxPush();
+#if VERSION == VERSION_DEMO
+    cMtx_YrotM(*calc_mtx, (cM_ssin((iVar2 + iVar4) * (REG10_S(3) - 800)) * (REG9_S(74) + 900)));
+#else
+    cMtx_YrotM(*calc_mtx, cM_ssin((iVar2 + iVar4) * -800) * 900.0f);
+#endif
+    MtxPosition(&sp5C, &sp68);
+    *pDst = sp68.normZC();
+    MtxPull();
 }
 
 /* 000006E8-00000D30       .text draw__21daMajuu_Flag_packet_cFv */
 void daMajuu_Flag_packet_c::draw() {
-    /* Nonmatching */
+    j3dSys.reinitGX();
+#if VERSION > VERSION_DEMO
+    GXSetNumIndStages(0);
+#endif
+    dKy_GxFog_tevstr_set(mpTevStr);
+    dKy_setLight_mine(mpTevStr);
+    GXClearVtxDesc();
+    GXSetVtxDesc(GX_VA_POS,GX_INDEX8);
+    GXSetVtxDesc(GX_VA_NRM,GX_INDEX8);
+    GXSetVtxDesc(GX_VA_TEX0,GX_INDEX8);
+    GXSetVtxAttrFmt(GX_VTXFMT0,GX_VA_POS, GX_CLR_RGBA,GX_F32, 0);
+    GXSetVtxAttrFmt(GX_VTXFMT0,GX_VA_NRM, GX_CLR_RGB,GX_F32, 0);
+    GXSetVtxAttrFmt(GX_VTXFMT0,GX_VA_TEX0, GX_CLR_RGBA,GX_F32, 0);
+    GXSetArray(GX_VA_POS, getPos(), sizeof(cXyz) * 1);
+    GXSetArray(GX_VA_NRM, getNrm(), sizeof(cXyz) * 1);
```

Comment:
Is `* 1` required to match?

Reply:
No, but since it's grabbing an array I wanted to show it's only using 1 element.